spec/03-plugins/10-key-auth/02-access_spec.lua

```diff
@@ -347,22 +362,49 @@ describe("Plugin: key-auth (access)", function()
 
     for type, _ in pairs(harness) do
       describe(type, function()
-        it("false sends key to upstream", function()
-          local res   = assert(client:send(harness[type][1]))
-          local body  = assert.res_status(200, res)
-          local json  = cjson.decode(body)
-          local field = type == "post_data.params" and json.post_data.params or json[type]
-          assert.equal("kong", field.apikey)
-        end)
-        it("true doesn't send key to upstream", function()
-          local res   = assert(client:send(harness[type][2]))
-          local body  = assert.res_status(200, res)
-          local json  = cjson.decode(body)
-          local field = type == "post_data.params" and json.post_data.params or json[type]
-          assert.is_nil(field.apikey)
-        end)
+        for _, content_type in pairs({
+          "application/x-www-form-urlencoded",
+          "application/json",
+          "multipart/form-data",
+          }) do
```

Comment:
Unfortunately this will produce invalid tests because our `client:send()` utility will mutate the given request (`opts.body = cjson.encode(...)` and co.)

Thus, the first run of this loop (we could use `ipairs` here but on my machine, it will be the first element inserted) will mutate the above test harness, and our `body` properties will not be Lua table, but strings (because of the `application/x-www-form-urlencoded` MIME type). Subsequent runs silently keep sending the form-urlencoded data since we give `opts.body` as already valid, urlencoded payload (this nullifying this loop's effect).

An easy fix would be to make this loop supersede the one iterating over the hardness (another good reason to do so it related to another comment)